### PR TITLE
Creating support for Jaleco JF11/JF14 (Mapper 140)

### PR DESCRIPTION
**Pull request by Shideravan**
Now the following games are supported in ares:

* Youkai Club: https://youtu.be/ol3mdaTrao8
* Mississippi Satsujin Jiken: https://youtu.be/9zmvD8aFvhE
* Bio Senshi Dan - Increaser Tono Tatakai: https://youtu.be/eiSniHKkfLI

There's a few graphical errors in _Bio Senshi Dan - Increaser Tono Tatakai_ (eg.: the title screen), but I'm unsure it's mapper related.

### CODE REVIEW

**Comment by LukeUsher** [comment]
I'm not sure this implementation is correct? 

This mapper doesn't support CHR RAM, and it doesn't support save ram either; yet your implementation defines/loads both, and implements CHR RAM.

See:
https://wiki.nesdev.com/w/index.php?title=INES_Mapper_140
http://bootgod.dyndns.org:7777/profile.php?id=3966
http://bootgod.dyndns.org:7777/profile.php?id=1548

**Comment by Shideravan** [comment]
Yes, you're right. Fixed!

File: ares/fc/cartridge/board/jaleco-jf11-jf14.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
+struct Jaleco_JF11_JF14 : Interface {
+  static auto create(string id) -> Interface* {
+    if(id == "JALECO-JF11-JF14") return new Jaleco_JF11_JF14;
+    return nullptr;
+  }
+
+  Memory::Readable<n8> programROM;
+  Memory::Writable<n8> programRAM;
```
Comment (LukeUsher) [inline ares/fc/cartridge/board/jaleco-jf11-jf14.cpp]: this mapper doesn't have programRAM?

File: ares/fc/cartridge/board/jaleco-jf11-jf14.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  Memory::Readable<n8> characterROM;
+
+  auto load() -> void override {
+    Interface::load(programROM, "program.rom");
+    Interface::load(programRAM, "save.ram");
```
Comment (LukeUsher) [inline ares/fc/cartridge/board/jaleco-jf11-jf14.cpp]: same here, no program ram

**Review by LukeUsher — CHANGES_REQUESTED** [review]

File: ares/fc/cartridge/board/jaleco-jf11-jf14.cpp (hunk 1 above)
Comment (Shideravan) [reply ares/fc/cartridge/board/jaleco-jf11-jf14.cpp]: removed

File: ares/fc/cartridge/board/jaleco-jf11-jf14.cpp (hunk 2 above)
Comment (Shideravan) [reply ares/fc/cartridge/board/jaleco-jf11-jf14.cpp]: Removed

File: ares/fc/cartridge/board/jaleco-jf11-jf14.cpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    if(characterROM) return characterROM.read(address);
+    return data;
+  }
+
+  auto writeCHR(n32 address, n8 data) -> void override {
+    if(address & 0x2000) {
+      address = address >> !mirror & 0x0400 | (n10)address;
+      return ppu.writeCIRAM(address, data);
+    }
+    address = characterBank << 13 | (n13)address;
```
Comment (LukeUsher) [inline ares/fc/cartridge/board/jaleco-jf11-jf14.cpp]: this check can go; address is not used after this so assigning/calculating a banked address is wasteful

**Review by LukeUsher — APPROVED** [review]

**Pull request merged**